### DISCUSSION TITLE
emmc installation won't boot without usb stick

### DISCUSSION THREAD

**Discussion by qooxzuub** [velvet-os/imagebuilder#262] [Q&A]
I have an Asus C201pa chromebook.  The write-protect screw is removed and gbb.flags is set to 0x00000019, with firmware id: Google_Veyron_Speedy.6588.160.0 (copied from the boot screen).

I am using this image on a USB stick:
https://github.com/hexdump0815/imagebuilder/releases/tag/231001-02

So I have Linux version 6.1.51-stb-cbr+ (root@armv7l) on Debian 12.2.0.

After setting up the USB stick for veyron/speedy, it booted fine.

Then I followed these instructions to sync to the emmc, which is /dev/mmcblk0:

https://github.com/hexdump0815/imagebuilder/blob/main/doc/chromebooks/installation/basic-installation.md

I used the workaround to get a functioning cgpt from oldstable.

When I now reboot with the USB stick removed, after the chromeos 'OS verification is off' screen, the screen goes black, then the screen flashes briefly  and then the machine reboots to the 'Chrome OS is missing or damaged' screen. I was hoping it would boot Debian.

When I reboot with the USB stick present and hit control-U, it boots Debian and mounts the correct partitions from /dev/mmcblk0 and uses one of them as /. No USB partitions are mounted. So I can eject the USB stick after booting and everything seems to work perfectly.

Any suggestions for getting booting working from /dev/mmcblk0 without the USB stick? 

(I don't really understand how the boot process is supposed to work. If there is something to read about that please let me know.)

**Comment by thenameisluk** [comment]
@qooxzuub apologies
when rewriting the guide i've failed to preserve correct procedure for veyron

step 6 should look like
```
# for veyron the following four lines should be used instead of the two above:
cgpt add -i 1 -t kernel -b 73728 -s 32768 -l KernelA -S 1 -T 2 -P 10 /dev/$disk
cgpt repair /dev/$disk
cgpt add -i 2 -t kernel -b 106496 -s 32768 -l KernelB -S 0 -T 2 -P 5 /dev/$disk
cgpt repair /dev/$disk
```

since beginning of veyron emmc is read-only

do the installation with ^ as step 6 and it should work
if you run into another issue let me know

i will submit fix as soon as i can

**Reply by qooxzuub** [reply]
Thank you! It works.

dd in step 9 complained about a full partition error, so I removed the bs=1024k parameter and it still complained. In each case, one record was not transferred (records_in = records_out + 1). Doesn't seem to matter anyhow :-)

**Reply by thenameisluk** [reply]
@qooxzuub glad to hear that it worked

when it comes to veyron you might be interested in [tuning firefox](https://github.com/hexdump0815/imagebuilder/blob/main/doc/firefox-tuning.md) to extend life of ur emmc
